### PR TITLE
torch.nn.functional.conv2d support

### PR DESCRIPTION
**Pull request by yuzhiyiliu**
It works fine in my program. If it doesn't work in your program, please feedback to me and I will continue to support it.

### CODE REVIEW

File: torch2trt/converters/conv2d.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    in_channels  = ctx.method_args[0].size()[1]
+    out_channels = ctx.method_args[1].size()[0]
+    kernel_size  = tuple(ctx.method_args[1].size()[2:4])
+    stride       = ctx.method_args[3]
+    padding      = ctx.method_args[4]
+    dilation     = ctx.method_args[5]
+    groups       = ctx.method_args[6]
+    bias = False if ctx.method_args[2] is None else True
+
+    module = torch.nn.Conv2d(
```
Comment (SrivastavaKshitij) [inline torch2trt/converters/conv2d.py:16]: module refers to mapping of native pytorch layer to Tensorrt layer. 

Refer to this documentation

https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-601/tensorrt-api/python_api/infer/Graph/Layers.html#iconvolutionlayer

nn.conv2d should map to iconvolution layer in trt

Comment (yuzhiyiliu) [reply torch2trt/converters/conv2d.py:16]: Very happy to receive your reply！
The torch.nn.Conv2d maps to iconvolution layer in trt, and it's mature.
I just wrapped torch.nn.functional.conv2d with torch.nn.Conv2d, they have the same effect, so I don't have to make wheels anymore.

**Comment by ElegantLin** [comment]
Hi, I try your method on Jetson Nano. Although there is no error, it would throw some warnings. After the program is finished, it will throw 'Segment Fault'. Do you know the reason?

Thanks a lot.

**Comment by yuzhiyiliu** [comment]
@ElegantLin 
Glad you used my code, please tell me more information.
e.g. which line of code went wrong?

**Comment by ElegantLin** [comment]
Hi, THX for your quick reply. I have no idea about which line of the code. However, I could give you more information and I hope it helps. I used `ipython`.

```
import torch
from torch2trt import torch2trt
from efficientnet_pytorch import EfficientNet
model = EfficientNet.from_pretrained('efficientnet-b3').eval().cuda()
x = torch.ones((1, 3, 224, 224)).cuda()
model_trt = torch2trt(model, [x])
```

And it gives a lot of warnings like
`Warning: Encountered known unsupported method torch.nn.functional.conv2d`

After I exited `ipython`, it will throw `Segmentation fault (core dumped)`. Do you know the reason?

Thanks a lot :)

**Comment by yuzhiyiliu** [comment]
@ElegantLin I suspect it did not find my plugin.

**Comment by ElegantLin** [comment]
Hi, @yuzhiyiliu 

Sorry, I did not use your plugin before. I encountered a problem when executing the same code above. The error is in Line 29 and Line 41

```
  File "/usr/local/lib/python3.6/dist-packages/efficientnet_pytorch/utils.py", line 144, in forward
    x = F.conv2d(x, self.weight, self.bias, self.stride, self.padding, self.dilation, self.groups)
  File "/torch2trt/torch2trt/torch2trt.py", line 218, in wrapper
    converter["converter"](ctx)
  File "/torch2trt/torch2trt/converters/conv2d.py", line 29, in convert_conv2d
    convert_Conv2d(ctx)
  File "/torch2trt/torch2trt/converters/Conv2d.py", line 41, in convert_Conv2d
    layer.stride = stride
TypeError: (): incompatible function arguments. The following argument types are supported:
    1. (arg0: tensorrt.tensorrt.IConvolutionLayer, arg1: tensorrt.tensorrt.DimsHW) -> None

Invoked with: <tensorrt.tensorrt.IConvolutionLayer object at 0x7f3cd21c70>, ([1, 1], [1, 1])
```

Thanks for your help.

**Comment by yuzhiyiliu** [comment]
@ElegantLin 
The stride mast be a int or tuple of two ints, ([1,1], [1,1]) -->(1,1)

**Comment by ElegantLin** [comment]
Hi, Thanks for your tips. I have made all ([1, 1], [1, 1]) to (1,1).

However, I met another problem. I set the log level as verbose to find more details. The other code is the same as above.

```
...
[TensorRT] VERBOSE: Layer(Pooling): (Unnamed Layer* 488) [Pooling], Tactic: -1, (Unnamed Layer* 487) [ElementWise]_output[Float(1536,8,10)] -> (Unnamed Layer* 488) [Pooling]_output[Float(1536,1,1)]
[TensorRT] VERBOSE: Layer(FullyConnected): (Unnamed Layer* 491) [Fully Connected], Tactic: 48, (Unnamed Layer* 488) [Pooling]_output[Float(1536,1,1)] -> (Unnamed Layer* 491) [Fully Connected]_output[Float(1000,1,1)]
[TensorRT] VERBOSE: Layer(Shuffle): (Unnamed Layer* 492) [Shuffle], Tactic: 0, (Unnamed Layer* 491) [Fully Connected]_output[Float(1000,1,1)] -> output_0[Float(500)]
Segmentation fault (core dumped)
```

It seems that it threw an error before it finished. Have you met this problem? Thanks a lot.

**Comment by nik1806** [comment]
@yuzhiyiliu Hi, Can you tell me how to use your plugins? 
I am getting the following error: 
efficientnet 2 tensorrt torch.nn.functional.conv2d not support #218

**Comment by wangjue-wzq** [comment]
@yuzhiyiliu Hi, I use your plugins, but get error

`
    def forward(self, input):
        """
        forward func
        :param input: input
        """

        out = F.conv2d(
            input,
            self.weight * self.scale,
            bias=self.bias,
            stride=self.stride,
            padding=self.padding,
            dilation=1,
            groups=1
        )
        return out

File "/usr/local/lib/python3.6/dist-packages/torch2trt-0.4.0-py3.6.egg/torch2trt/converters/conv2d.py", line 11, in convert_conv2d
 stride       = ctx.method_args[3]
IndexError: tuple index out of range `

**Pull request closed without merging**